lib/matplotlib/contour.py

```diff
@@ -964,7 +964,7 @@ def __init__(self, ax, *args, **kwargs):
                     segs,
                     antialiaseds=aa,
                     linewidths=width,
-                    linestyle=[lstyle],
+                    linestyles=[lstyle],
```

Comment:
This is from https://github.com/tacaswell/matplotlib/commit/154cd87cca035ef040f2370709726ccd508f260c and is intentional so that the kwarg is handled by the named keyword in the `__init__` signature rather than falling through the kwargs to the `set_*` magic. 

This should go in either way but I can pull it out of this PR if you want.